waspc/src/Wasp/Generator/Job/IO.hs

```diff
@@ -29,9 +33,26 @@ readJobMessagesAndPrintThemPrefixed chan = runPrefixedWriter go
         J.JobOutput {} -> printJobMessagePrefixed jobMsg >> go
         J.JobExit {} -> return ()
 
+collectJobTextOutputUntilExitReceived :: Chan J.JobMessage -> IO [Text]
+collectJobTextOutputUntilExitReceived chan = collectJobMessages chan <&> mapMaybe jobMessageToText
+  where
+    jobMessageToText :: J.JobMessageData -> Maybe Text
+    jobMessageToText = \case
+      J.JobOutput text _ -> Just text
+      J.JobExit _ -> Nothing
+
 printJobMessage :: J.JobMessage -> IO ()
 printJobMessage jobMsg = do
   let outHandle = getJobMessageOutHandle jobMsg
   let message = getJobMessageContent jobMsg
   T.IO.hPutStr outHandle message
   hFlush outHandle
+
+collectJobMessages :: Chan J.JobMessage -> IO [J.JobMessageData]
+collectJobMessages = collect []
+  where
+    collect messages chan = do
+      jobMsg <- readChan chan
+      case J._data jobMsg of
+        J.JobExit {} -> return messages
+        message -> collect (message : messages) chan
```

Comment:
Btw I can't remember what is the difference between JobMessage and JobMessageData, but this function says it collects JobMessages but it returns JobMessageData, is that an issue? Should the name also have `Data` in it?

Reply:
Aha ok, I realized it now from the casing in another function above. So JobMessageData is either Text or Exit? In that case, since you know you are returning only Text's here, why not return `IO [Text]`? It is more precise!